### PR TITLE
fix infer args

### PR DESCRIPTION
**Pull request by liangxhao**
Thank you for your contribution to the MindOCR repo.
Before submitting this PR, please make sure:

- [x] You have read the [Contributing Guidelines on pull requests](https://github.com/mindspore-lab/mindcv/blob/main/CONTRIBUTING.md)
- [x] Your code builds clean without any errors or warnings
- [x] You are using approved terminology
- [ ] You have added unit tests

## Motivation

简化接口参数，和训练端对齐

det_model_name/det_config_path  -> det_model_name_or_config
cls_model_name/cls_config_path     -> cls_model_name_or_config
rec_model_name/rec_config_path   ->  rec_model_name_or_config

对于推理接口参数det_model_name和det_config_path，前者传入模型名，后者传入模型yaml路径，两者是关联对应的

用户在使用时二选一，用以确定预后处理参数，导致易用性低
现在合二为一改成一个参数det_model_name_or_config，和训练侧对齐，同时支持传入模型名和yaml路径

## Test Plan

(How should this PR be tested? Do you require special setup to run the test or repro the fixed bug?)

## Related Issues and PRs

(Is this PR part of a group of changes? Link the other relevant PRs and Issues here. Use https://help.github.com/en/articles/closing-issues-using-keywords for help on GitHub syntax)

### CODE REVIEW

**Review by HaoyangLee — APPROVED** [review]

File: deploy/py_infer/src/utils/adapted/__init__.py [hunk 1]
```diff
@@ -1,18 +1,41 @@
...
+        filename = os.path.abspath(os.path.join(MINDOCR_CONFIG_PATH, MINDOCR_MODELS[model_name_or_config]))
+    elif model_name_or_config in PADDLEOCR_MODELS:
+        filename = os.path.abspath(os.path.join(PADDLEOCR_CONFIG_PATH, PADDLEOCR_MODELS[model_name_or_config]))
+    elif model_name_or_config in MMOCR_MODELS:
+        filename = os.path.abspath(os.path.join(MMOCR_CONFIG_PATH, MMOCR_MODELS[model_name_or_config]))
     else:
-        raise ValueError(f"The model name {name} is not supported, please check the model name.")
+        raise ValueError(
+            f"The {model_name_or_config} must be a model name or YAML config file, "
+            "please check if the file exists, or if model name in the support list."
```
[the remark this answers is not in the document]
Comment (liangxhao) [reply deploy/py_infer/src/utils/adapted/__init__.py]: 已修改

File: docs/en/inference/inference_tutorial_en.md [hunk 2]
```diff
@@ -161,38 +162,31 @@ Enter the inference directory：`cd deploy/py_infer`.
...
 Notes：
 
-1. For the adapted model, `*_model_path`,`*_model_name` and `*_config_path` are bounded together, please refer to
-   [MindOCR Models Support List](./models_list_en.md) and
-   [Third-Party Models Support List](./models_list_thirdparty_en.md). Both `*_model_name` and `*_config_path` are used
-   to determine pre/post processing parameters, and you can choose which one to use;
-2. If you need to adapt to your own model, `*_config_path` can simply pass in your own yaml file. Please refer to the
-   [configs](../../../configs) of the MindOCR model or the [configs](../../../deploy/py_infer/src/configs) of
-   third-party models for file format.
+  `*_model_name_or_config` can use the model name or YAML config file, please refer to [MindOCR Models Support List](./models_list_en.md) and
```
[the remark this answers is not in the document]
Comment (liangxhao) [reply docs/en/inference/inference_tutorial_en.md]: 已修改

File: docs/en/inference/inference_tutorial_en.md [hunk 3]
```diff
@@ -161,38 +162,31 @@ Enter the inference directory：`cd deploy/py_infer`.
...
 
-| name            | type | default | description                    |
-|:----------------|:-----|:--------|:-------------------------------|
-| det_model_path  | str  | None    | Model path for text detection  |
-| det_model_name  | str  | None    | Model name for text detection  |
-| det_config_path | str  | None    | Config file for text detection |
+| name                     | type | default | description                                       |
+|:-------------------------|:-----|:--------|:--------------------------------------------------|
+| det_model_path           | str  | None    | Model path for text detection                     |
+| det_model_name_or_config | str  | None    | Model name or YAML config file for text detection |
```
[the remark this answers is not in the document]
Comment (liangxhao) [reply docs/en/inference/inference_tutorial_en.md]: 3处都已修改

**Review by heepengpeng — APPROVED** [review]

File: deploy/py_infer/src/utils/adapted/__init__.py [hunk 4]
```diff
@@ -1,18 +1,41 @@
...
+    elif model_name_or_config in MINDOCR_MODELS:
+        filename = os.path.abspath(os.path.join(MINDOCR_CONFIG_PATH, MINDOCR_MODELS[model_name_or_config]))
+    elif model_name_or_config in PADDLEOCR_MODELS:
+        filename = os.path.abspath(os.path.join(PADDLEOCR_CONFIG_PATH, PADDLEOCR_MODELS[model_name_or_config]))
+    elif model_name_or_config in MMOCR_MODELS:
+        filename = os.path.abspath(os.path.join(MMOCR_CONFIG_PATH, MMOCR_MODELS[model_name_or_config]))
     else:
-        raise ValueError(f"The model name {name} is not supported, please check the model name.")
+        raise ValueError(
+            f"The {model_name_or_config} must be a model name or YAML config file, "
```
Comment (HaoyangLee) [inline deploy/py_infer/src/utils/adapted/__init__.py:23]:
```suggestion
            f"The {model_name_or_config} must be a model name or YAML config file path, "
```

**Review by HaoyangLee — APPROVED** [review]

File: deploy/py_infer/src/utils/adapted/__init__.py (hunk 4 above)
Comment (liangxhao) [reply deploy/py_infer/src/utils/adapted/__init__.py]: 已修改

File: deploy/py_infer/src/infer_args.py [hunk 5]
```diff
@@ -43,16 +43,19 @@ def get_args():
     )
 
     parser.add_argument("--det_model_path", type=str, required=False, help="Detection model file path.")
-    parser.add_argument("--det_model_name", type=str, required=False, help="Detection model name.")
-    parser.add_argument("--det_config_path", type=str, required=False, help="Detection config file path.")
+    parser.add_argument(
+        "--det_model_name_or_config", type=str, required=False, help="Detection model name or config file path."
```
Comment (VictorHe-1) [inline deploy/py_infer/src/infer_args.py:47]: 这个不是必须参数吗？required=True

**Review by VictorHe-1 — APPROVED** [review]

**Review by jianyunchao — APPROVED** [review]

**Pull request merged**